### PR TITLE
[coq] Adapt to coq/coq#6454 "forbid calling to_constr on open terms"

### PR DESCRIPTION
**Pull request by ejgallego**
This PR is IMO interesting and seems to suggest quite a few Coq API modifications that are a must.
    
In particular #6310 seems like one of the most flagrant omissions from the EConstr API.

### CODE REVIEW

**Comment by ppedrot** [comment]
The PR has made it into Coq. Can you merge this one?

**Comment by maximedenes** [comment]
This was merged to the wrong branch and breaks Coq's 8.8 CI. Please fix ASAP, given how close we are to the release.

**Comment by Zimmi48** [comment]
Shit @mattam82, didn't you see what I wrote in coq/coq#7205? Your `8.8+alpha` branch is now incompatible with Coq 8.8 and consequently our CI for `v8.8` is going to be broken.

**Comment by Zimmi48** [comment]
FTR the solution is to create a new branch 8.9+alpha from this branch, and to revert this PR in 8.8+alpha only.

**Comment by ejgallego** [comment]
In some version of the CI documents, a standard branch naming for CI contribs was suggested, maybe it is time we start to enforce it and thus submit issues to all devs upstream. However, this may be a delicate issue.

**Comment by Zimmi48** [comment]
I think that the naming scheme is not the problem here.

**Comment by ejgallego** [comment]
Indeed, sorry for the mistake, let me rephrase my statement "a standard branch **layout** for CI contribs was suggested".
